### ISSUE TITLE
Update Jakarta Servlet chapter for Servlet 6.0 (part of Jakarta EE 10)

### ISSUE THREAD

**Issue by kito99** [jakartaee/jakartaee-tutorial#74]
Current tutorial chapters for Servlet: https://jakartaee.github.io/jakartaee-documentation/jakartaee-tutorial/current/web/webapp/webapp.html

For background, here are the changes for Servlet 6.0:

* Article: https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/Top-5-things-to-know-about-the-Jakarta-Servlet-60-API-release
* Release notes: https://jakarta.ee/specifications/servlet/6.0/

The only change that may affect the tutorial is the removal of deprecated features.

Please do the following:

* Review the examples and look for any deprecated methods. If you see one, let us know so we can create a ticket to fix the example, because it won't compile in Servlet 6. (Alternatively, if you have the interest and bandwidth to update the examples that build and run in the https://github.com/eclipse-ee4j/jakartaee-examples repo, let us know and we'll create a separate ticket for it.)
* Add the following to the https://jakartaee.github.io/jakartaee-documentation/jakartaee-tutorial/current/web/servlets/servlets.html#_further_information_about_jakarta_servlet_technology section:
  * Add a NOTE section that explains that previously deprecated methods are now removed in Servlet 6.0, and link to the list of methods: https://jakarta.ee/specifications/servlet/5.0/apidocs/deprecated-list.html
  * Add a link to the Servlet 6 release notes: https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/Top-5-things-to-know-about-the-Jakarta-Servlet-60-API-release
  * Update the spec URL to point to the Servlet 6 spec: https://jakarta.ee/specifications/servlet/6.0/.

**Comment by kito99** [comment]
@markt-asf and @stuartwdouglas can you take a look at this in the next couple of days? I'd love to know (a) if this looks accurate for Jakarta EE 10 and (b) if you or someone on the team can make this change.

**Comment by jhammen** [comment]
happy to work on this but it looks like there are some major changes to these sections still waiting in PR #36 @kito99 @BalusC  do you think these changes are likely to be merged soon?

**Comment by kito99** [comment]
@jhammen I think the changes to that PR are good; @BalusC can you merge in `main` and test the build (if you haven't already)?

**Comment by BalusC** [comment]
Done. You can advance.

**Comment by jhammen** [comment]
First example mentioned is "hello-servlet" which does not exist at the path given, looks like it was repaced with a "hello2" example which has similar code and an additional servlet. So the copy needs to be updated, code snippets should use the new code and links should point to the new example paths.

The "hello2" example does not appear to have any deprecated methods, works fine with tomcat:10.1.16

**Comment by BalusC** [comment]
It's the other way round. `hello1` and `hello2` from previous versions have been replaced by `hello-faces` and `hello-servlet`. Surely the old ones will work fine but they expose outdated/bad practices these days.

**Comment by BalusC** [comment]
Oh wait. They're already created. The PR is still pending review/merge. https://github.com/eclipse-ee4j/jakartaee-examples/pull/80

**Comment by jhammen** [comment]
OK that all makes sense and I'm glad this already fixed pending PR.

@BalusC  I don't want to get in the way if you have this update under control, I'm happy to step back. Did you want to assign me tickets for things you intended to do? Otherwise maybe @kito99  can assign other sections for me to work while I have the time to spare

**Comment by jhammen** [comment]
> @jhammen you can update the text in the meantime if you want..

You're right - the 6.0 links are now added/updated, please see the above commit reflected here:

https://jhammen.github.io/jeetutorial/current/web/servlets/servlets.html#_further_information_about_jakarta_servlet_technology

**Comment by BalusC** [comment]
https://github.com/eclipse-ee4j/jakartaee-examples/pull/80 was finally merged last June.

**Comment by kito99** [comment]
Cool! Thanks, @BalusC. @jhammen do you wnt to pick this back up again?

**Comment by jhammen** [comment]
For sure, will take a look this week

**Comment by kito99** [comment]
thanks @jhammen !

**Comment by jhammen** [comment]
> Review the examples and look for any deprecated methods. If you see one, let us know so we can create a ticket to fix the example, because it won't compile in Servlet 6. 

I've reviewed and tested the servlet examples mentioned in this chapter and found no issues with deprecated methods.

Examples tested were:

* hello-servlet
* hello-faces
* mood
* dukeeft
* fileupload

There are some smaller snippets that are not official examples e.g. TimeZoneHandler, ShutdownExample, these also look free of deprecated or removed methods.

**Comment by jhammen** [comment]
@kito99  please check above PR with content changes. I'm not seeing much else to add or change for Servlet 6.0 so let me know if there's anything left for this issue

(BTW it's already a pretty long chapter, may want to chop it up into more than one page at some point.)
